ambry-rest/src/test/java/com.github.ambry.rest/NettyServerFactoryTest.java

```diff
@@ -25,77 +30,88 @@
  * Tests functionality of {@link NettyServerFactory}.
  */
 public class NettyServerFactoryTest {
+  // dud properties. server should pick up defaults
+  private static final RestRequestHandler REST_REQUEST_HANDLER = new MockRestRequestResponseHandler();
+  private static final PublicAccessLogger PUBLIC_ACCESS_LOGGER = new PublicAccessLogger(new String[]{}, new String[]{});
+  private static final RestServerState REST_SERVER_STATE = new RestServerState("/healthCheck");
+  private static final SSLFactory SSL_FACTORY = RestTestUtils.getTestSSLFactory();
 
   /**
    * Checks to see that getting the default {@link NioServer} (currently {@link NettyServer}) works.
    */
   @Test
-  public void getNettyServerTest() {
-    // dud properties. server should pick up defaults
+  public void getNettyServerTest() throws Exception {
     Properties properties = new Properties();
-    VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
-    RestRequestHandler restRequestHandler = new MockRestRequestResponseHandler();
-    PublicAccessLogger publicAccessLogger = new PublicAccessLogger(new String[]{}, new String[]{});
-    RestServerState restServerState = new RestServerState("/healthCheck");
+    doGetNettyServerTest(properties, SSL_FACTORY);
+    doGetNettyServerTest(properties, null);
+    properties.setProperty("netty.server.enable.ssl", "true");
+    doGetNettyServerTest(properties, SSL_FACTORY);
+  }
 
+  /**
+   * Test that a {@link NettyServer} can be constructed by the factory.
+   * @param properties the {@link Properties} to use.
+   * @param sslFactory the {@link SSLFactory} to use.
+   */
+  private void doGetNettyServerTest(Properties properties, SSLFactory sslFactory) {
+    VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
+    NettyConfig nettyConfig = new NettyConfig(verifiableProperties);
     NettyServerFactory nettyServerFactory =
-        new NettyServerFactory(verifiableProperties, new MetricRegistry(), restRequestHandler, publicAccessLogger,
-            restServerState);
+        new NettyServerFactory(verifiableProperties, new MetricRegistry(), REST_REQUEST_HANDLER, PUBLIC_ACCESS_LOGGER,
+            REST_SERVER_STATE, sslFactory);
     NioServer nioServer = nettyServerFactory.getNioServer();
     assertNotNull("No NioServer returned", nioServer);
     assertEquals("Did not receive a NettyServer instance", NettyServer.class.getCanonicalName(),
         nioServer.getClass().getCanonicalName());
+    Map<Integer, ChannelInitializer<SocketChannel>> channelInitializers = nettyServerFactory.getChannelInitializers();
```

Comment:
minor: instead of introducing a function, i would prefer the variable being package private.